scripts/use-yarn-please.js

```diff
@@ -0,0 +1,58 @@
+const path = require('path');
+const { spawnSync } = require('child_process');
+const npmPath = process.env.npm_execpath;
+
+const Strings = {
+  useYarnInstead: `Looks like you are trying to run "npm install", this repository has migrated to use Yarn as its package manager for installations.
+  `,
+  yarnDetected: `Great! Looks like you already have yarn installed!
+  `,
+  installYarn: `You current do not have an installation of Yarn in your PATH. Be sure to install the latest stable version of Yarn here:
+
+Download an installer here: https://yarnpkg.com/en/docs/install
+
+Windows users can run the installer here:
+
+  https://yarnpkg.com/latest.msi
+
+Mac users can run a one-liner for the install:
+
+  curl -o- -L https://yarnpkg.com/install.sh | bash
+`,
```

Comment:
Why no mention of `npm install yarn`? At least on the surface it seems to work okay, and it's simpler.

Reply:
The answer is surprising - and is a recommendation against doing that by the yarn team themselves: https://yarnpkg.com/lang/en/docs/install/#alternatives-stable

>Note: Installation of Yarn via npm is generally not recommended. When installing Yarn with Node-based package managers, the package is not signed, and the only integrity check performed is a basic SHA1 hash, which is a security risk when installing system-wide apps.

>For these reasons, it is highly recommended that you install Yarn through the installation method best suited to your operating system.

Reply:
As we discussed in person earlier, I think their reasoning is kind of silly since the exact same concern applies with any other npm package. Maybe we could just say something like:

``Please install the latest stable version of Yarn following the instructions at https://yarnpkg.com/en/docs/install or by running `npm install -g yarn`.``